fggs/indices.py

```diff
@@ -0,0 +1,1293 @@
+"""
+# Algebraic index types
+
+A tensor dimension often corresponds to a random variable, and indices along
```

Comment:
I try to avoid the word "dimension" because it has two meanings: a 3x3 matrix, considered as a multidimensional array, has 2 dimensions, but in a linear algebra context, it is a linear mapping on a vector space with 3 dimensions (or could itself be thought of as a vector with 9 dimensions!)

Reply:
I totally agree so what **is** the word that people have settled on for the thing that's counted by `ndim`?? PyTorch documentation says "dimension" :-/

Reply:
Ah you use the word "axis"

Reply:
NumPy inconsistently uses both "axis" and "dimension"; PyTorch uses "dimension." Other terms are "index", "mode", or "way". For the named tensor notation paper, we used "axis".